pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -338,7 +338,38 @@ public CompletableFuture<Void> initialize() {
             if (cursor.getName().startsWith(replicatorPrefix)) {
                 String localCluster = brokerService.pulsar().getConfiguration().getClusterName();
                 String remoteCluster = PersistentReplicator.getRemoteCluster(cursor.getName());
-                futures.add(addReplicationCluster(remoteCluster, cursor, localCluster));
+                final CompletableFuture<Void> future = new CompletableFuture<>();
+                addReplicationCluster(remoteCluster, cursor, localCluster).whenComplete((__, e) -> {
+                    if (e == null) {
+                        future.complete(null);
+                    } else {
+                        Throwable throwable = e;
+                        while (throwable.getCause() != null) {
+                            throwable = throwable.getCause();
+                        }
+                        if (throwable instanceof MetadataStoreException.NotFoundException
+                                && throwable.getMessage().equals(remoteCluster)) {
```

Comment:
I feel to make `ex.getMessage` a signal to remove the cursor is not a safe solution, can we "check the Namespace Policy & Topic Policy" instead of checking `ex.getMessage`?

Reply:
I checked the `ex.getMessage` just to catch a specific failure:

```java
        return replicationClients.computeIfAbsent(cluster, key -> {
            try {
                ClusterData data = clusterDataOp
                        .orElseThrow(() -> new MetadataStoreException.NotFoundException(cluster));
```

The exception message is always the same with the cluster name. If the behavior changed, this test would fail.

For other exceptions, I think we should not ignore them.

Reply:
Suggestion a logic like this:

```java
if (checkReplicationStillEnabledByPolicies()){
    deleteCursor();
    return;
}

startReplication(); // Don't ignore any exceptions here.
```

In this way, the exception `MetadataStoreException.NotFoundException` will not throw anymore, and it is easy to maintain (we do not care about `ex.message` even if others changed it).

Reply:
Sorry I didn't get the logic here. Did you mean only removing the cursor when the remote cluster is still a part of `topicPolicies`?

Reply:
> Sorry I didn't get the logic here. Did you mean only removing the cursor when the remote cluster is still a part of topicPolicies?

do remove the cursor only the remote cluster not exists in the `topicPolicies.replicationClusters`